### PR TITLE
[ISSUE #4979] Fix flaky integration test NormalMessageSendAndRecvIT

### PR DESCRIPTION
**Pull request by lizhanhui**
**Make sure to set the target branch to `develop`**

## What is the purpose of the change

Fix https://github.com/apache/rocketmq/issues/4979

Fix netty warning; https://blog.csdn.net/diaobatian/article/details/90696757
Fix naming typo of batch unregistration service

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java [hunk 1]
```diff
@@ -198,12 +198,11 @@ public PullResult getMessage(String group, String topic, int queueId, long offse
                     break;
                 case NO_MATCHED_MESSAGE:
                     pullStatus = PullStatus.NO_MATCHED_MSG;
-                    POP_LOGGER.warn("no matched message. GetMessageStatus={}, topic={}, groupId={}, requestOffset={}",
+                    POP_LOGGER.info("no matched message. GetMessageStatus={}, topic={}, groupId={}, requestOffset={}",
```
Comment (zhouxinyu) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java:201]: `NO_MATCHED_MESSAGE` means the pulled messages are filtered, so I think `info` is enough, maybe debug is a better level.

File: test/src/test/java/org/apache/rocketmq/test/smoke/NormalMessageSendAndRecvIT.java [hunk 2]
```diff
@@ -57,13 +60,33 @@ public void setUp() throws Exception {
...
+        while (true) {
+            try {
+                consumeStats = defaultMQAdminExt.examineConsumeStats(group);
+                messageQueueList = producer.getProducer().fetchPublishMessageQueues(topic);
+                if (!messageQueueList.isEmpty() && null != consumeStats
+                    && consumeStats.getOffsetTable().keySet().containsAll(messageQueueList)) {
+                    break;
+                } else {
+
+                    TimeUnit.MILLISECONDS.sleep(100);
```
Comment (aaron-ai) [inline test/src/test/java/org/apache/rocketmq/test/smoke/NormalMessageSendAndRecvIT.java:81]: Could we use awaitility to replace the sleep here?

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java [hunk 3]
```diff
@@ -225,7 +224,11 @@ public PullResult getMessage(String group, String topic, int queueId, long offse
                 getMessageResult.getMaxOffset(), foundList);
 
         } else {
-            POP_LOGGER.error("get message from store return null. topic={}, groupId={}, requestOffset={}", topic, group, offset);
+            long maxQueueOffset = brokerController.getMessageStore().getMaxOffsetInQueue(topic, queueId);
```
Comment (zhouxinyu) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java:227]: Log always if a null result is returned. I think this is an exceptional case. For the typical case, a clear result should be returned, even if `offset > maxOffset`(OFFSET_OVERFLOW_BADLY is returned).

File: remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingServer.java [hunk 4]
```diff
@@ -206,7 +206,7 @@ public Thread newThread(Runnable r) {
                 .channel(useEpoll() ? EpollServerSocketChannel.class : NioServerSocketChannel.class)
                 .option(ChannelOption.SO_BACKLOG, 1024)
                 .option(ChannelOption.SO_REUSEADDR, true)
-                .option(ChannelOption.SO_KEEPALIVE, false)
+                .childOption(ChannelOption.SO_KEEPALIVE, false)
```
Comment (zhouxinyu) [inline remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingServer.java:209]: The default value of childOption is true or false?

Comment (lizhanhui) [reply remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingServer.java:209]: Default tcp-ping is on. As we have application-level heartbeats, it is not required.

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java (hunk 3 above)
Comment (lizhanhui) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java:227]: At present, a clean start-up would observe bulks of errors in the server. This is unwanted and intimidating to new users. IMO, it's good to log conditionally and add potential ones if there is a need.

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java (hunk 1 above)
Comment (lizhanhui) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PopReviveService.java:201]: OK, let's downgrade it to debug.

File: test/src/test/java/org/apache/rocketmq/test/smoke/NormalMessageSendAndRecvIT.java (hunk 2 above)
Comment (lizhanhui) [reply test/src/test/java/org/apache/rocketmq/test/smoke/NormalMessageSendAndRecvIT.java:81]: It's virtually the same busy polling way.

**Review by aaron-ai — DISMISSED** [review]

**Review by aaron-ai — APPROVED** [review]

**Pull request merged**